jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildDockerMojo.java

```diff
@@ -100,42 +97,19 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
     MojoCommon.checkUseCurrentTimestampDeprecation(this);
 
+    MavenSettingsProxyProvider.activateHttpAndHttpsProxies(
+        getSession().getSettings(), getSettingsDecrypter());
+
+    MavenProjectProperties projectProperties =
+        MavenProjectProperties.getForProject(getProject(), getSession(), getLog());
     try {
-      RawConfiguration mavenRawConfiguration = new MavenRawConfiguration(this);
-      MavenProjectProperties projectProperties =
-          MavenProjectProperties.getForProject(getProject(), getSession(), getLog());
-
-      PluginConfigurationProcessor pluginConfigurationProcessor =
-          PluginConfigurationProcessor.processCommonConfigurationForDockerDaemonImage(
-              mavenRawConfiguration,
-              new MavenSettingsServerCredentials(
-                  getSession().getSettings(), getSettingsDecrypter()),
-              projectProperties,
-              dockerExecutable,
-              getDockerClientEnvironment(),
-              new MavenHelpfulSuggestions(HELPFUL_SUGGESTIONS_PREFIX));
-      MavenSettingsProxyProvider.activateHttpAndHttpsProxies(
-          getSession().getSettings(), getSettingsDecrypter());
-
-      ImageReference targetImageReference = pluginConfigurationProcessor.getTargetImageReference();
-
-      Path buildOutput = Paths.get(getProject().getBuild().getDirectory());
-
-      try {
-        JibBuildRunner.forBuildToDockerDaemon(targetImageReference, getTargetImageAdditionalTags())
-            .writeImageDigest(buildOutput.resolve("jib-image.digest"))
-            .writeImageId(buildOutput.resolve("jib-image.id"))
-            .build(
-                pluginConfigurationProcessor.getJibContainerBuilder(),
-                pluginConfigurationProcessor.getContainerizer(),
-                projectProperties::log,
-                new MavenHelpfulSuggestions(HELPFUL_SUGGESTIONS_PREFIX));
-
-      } finally {
-        // TODO: This should not be called on projectProperties.
-        projectProperties.waitForLoggingThread();
-        getLog().info("");
-      }
+      PluginConfigurationProcessor.runJibForDockerDaemonImage(
```

Comment:
I think this is generally nicer and cleaner, althought it seems weir'd to run the actual build from a static method defined in `PluginConfigurationProcessor` ?

Reply:
I restored the original method name (`process...ForDockerDaemonImage`) and made it return a configured `JibBuildRunner` instance, on which this plugin code will call `runBuild()` to kick the tire.